Scripts/aws_stack_provision.ps1

```diff
@@ -1,103 +1,302 @@
 ## Powershell script for
 
 param (
-    [string] $lansa_version
+[parameter(Mandatory=$true)]    [string] $lansa_version,
+[parameter(Mandatory=$false)]   [string] $tag_lansa_version ## lansa version that would be used to tag cloned stack
+
 )
 
```

Comment:
Rename $tag_lansa_version as $clone_lansa_version.
We said the very first line needs to be:
```
if ( [string]::IsNullOrWhiteSpace( $clone_lansa_version) ) {
   $tag_lansa_version = $lansa_version
}
```

Reply:
@robe070 So the script needs to do both provision new stack and clone the stack at once or it would be doing one process at a time?

Reply:
Thats not the best way to look at it.
The only difference between a Clone and not a Clone is that the Clone uses a previous version's saved state. So its just about where it finds the AMI and the database backups. Thats the only difference between Clone and not a Clone.
For a Clone it needs to know what to Clone it from. When its not a Clone its using ITS OWN SAVED STATE. Hence $clone_lansa_Version is set to $lansa_Version. That is, when a saved state is required use $lansa_version value (which you will refer to using $clone_lansa_Version so the code is simpler)